stm32-modules/heater-shaker/include/heater-shaker/heater_task.hpp

```diff
@@ -82,72 +137,129 @@ requires MessageQueue<QueueImpl<Message>, Message> class HeaterTask {
+                if (state.error_bitmap == 0) {
+                    state.system_status = State::IDLE;
+                }
+            }
+        }
+    }
+
+    auto visit_conversion(thermistor_conversion::Error error,
+                          TemperatureSensor& sensor) -> void {
+        switch (error) {
+            case thermistor_conversion::Error::OUT_OF_RANGE_LOW: {
+                sensor.temp_c = 0;
+                sensor.error = sensor.short_error;
             }
             case thermistor_conversion::Error::OUT_OF_RANGE_HIGH: {
-                auto message =
-                    messages::HostCommsMessage(messages::ErrorMessage{
-                        .code = errors::ErrorCode::
-                            HEATER_THERMISTOR_OUT_OF_RANGE_HIGH});
-                static_cast<void>(
-                    task_registry->comms->get_message_queue().try_send(
-                        message));
-                break;
+                sensor.temp_c = 0;
+                sensor.error = sensor.disconnected_error;
             }
         }
-        return 0;
     }
 
-    auto visit_conversion(double value) -> double { return value; }
+    auto visit_conversion(double value, TemperatureSensor& sensor) -> void {
+        if (value > sensor.overtemp_limit_c) {
+            sensor.error = sensor.overtemp_error;
+        } else {
+            sensor.error = errors::ErrorCode::NO_ERROR;
+        }
+        sensor.temp_c = value;
+    }
+
+    [[nodiscard]] auto most_relevant_error() const -> errors::ErrorCode {
+        // We have a lot of different errors from a lot of different sources.
+        // Sometimes more than one can occur at the same time; sometimes, that
+        // means that one has caused the other. We want to track them
+        // separately, but we also sometimes want to respond with just one error
+        // condition that sums everything up. This method is used by code that
+        // wants the single most relevant code for the current error condition.
+        if ((state.error_bitmap & State::SENSE_ERROR) != 0) {
+            // Prefer sense errors since they'll be most specific
+            if ((state.error_bitmap & State::PAD_SENSE_ERROR) != 0) {
+                // Prefer pad a errors to pad b errors arbitrarily
+                if ((state.error_bitmap & State::PAD_A_SENSE_ERROR) != 0) {
+                    return pad_a.error;
+                }
+                return pad_b.error;
+            }
+            return board.error;
+        }
+        return errors::ErrorCode::NO_ERROR;
+    }
     Queue& message_queue;
     tasks::Tasks<QueueImpl>* task_registry;
-    const thermistor_conversion::Conversion pad_a_conversion;
-    const thermistor_conversion::Conversion pad_b_conversion;
-    const thermistor_conversion::Conversion board_conversion;
-    int16_t pad_a_temperature;
-    int16_t pad_b_temperature;
-    int16_t board_temperature;
+    TemperatureSensor pad_a;
+    TemperatureSensor pad_b;
```

Comment:
Are these members public?

Reply:
no, the `private` section isn't in the diff but it's on line 134 of this file